scripts/production/patch_database.pl

```diff
@@ -0,0 +1,128 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+=head1 NAME
+
+patch_database.pl
+
+=head1 DESCRIPTION
+
+This script is a wrapper around ensembl's schema_patcher.pl to run it
+taking connection details from a production_reg_conf file.
+
+=head1 SYNOPSIS
+
+  perl patch_database.pl $COMPARA_REG compara_prev
+
+=head1 ARGUMENTS
+
+=head2 DATABASE SETUP
+
+=over
+
+=item B<[--url mysql://user[:passwd]@host[:port]/dbname]>
+
+URL of the database to test.
+
+=item B<[--reg_conf registry_configuration_file]>
+
+The Bio::EnsEMBL::Registry configuration file. If none given and no URL is
+given, the one set in ENSEMBL_REGISTRY will be used if defined, if not
+~/.ensembl_init will be used.
+
+=item B<[--reg_type reg_type]>
+
+The "type" or "group" under which the database is to be found in the Registry.
+
+=item B<[--reg_alias|--reg_name name]>
+
+The name or "species" under which the database is to be found in the Registry.
+
+=back
+
+=head2 SCHEMA PATCHER SETUP
+
+=over
+
+=item B<[--schema_patcher /path/to/schema_patcher.pl]>
+
+The path to schema_patcher.pl. If not given a default will be
+formed using the ENSEMBL_CVS_ROOT_DIR environment variable.
+
+=back
+
+=cut
+
+use strict;
+use warnings;
+
+use Getopt::Long;
+
+use Bio::EnsEMBL::Registry;
+
+use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
+use Bio::EnsEMBL::Compara::Utils::Registry;
+
+my ($url, $reg_conf, $reg_type, $reg_alias);
+my ($schema_patcher);
+
+# Arguments parsing
+GetOptions(
+    'url=s'                         => \$url,
+    'reg_conf|regfile|reg_file=s'   => \$reg_conf,
+    'reg_type=s'                    => \$reg_type,
+    'reg_alias|regname|reg_name=s'  => \$reg_alias,
+    'schema_patcher=s'              => \$schema_patcher,
+) or die "Error in command line arguments\n";
+
+if (@ARGV) {
+    die "ERROR: There are invalid arguments on the command-line: ". join(" ", @ARGV). "\n";
+}
+
+unless ($url or ($reg_conf and $reg_alias)) {
+    print "\nNeither --url nor --reg_conf and --reg_alias are defined. Some of those are needed to refer to the database being tested\n\n";
+    exit 1;
+}
+
+if ($reg_conf) {
+    Bio::EnsEMBL::Registry->load_all($reg_conf);
+}
+my $dba = $url
+    ? Bio::EnsEMBL::Compara::DBSQL::DBAdaptor->new( -URL => $url )
+    : Bio::EnsEMBL::Registry->get_DBAdaptor( $reg_alias, $reg_type || 'compara' );
+
+unless ($schema_patcher) {
+    die "Need to give the --schema_patcher option or set the ENSEMBL_CVS_ROOT_DIR environment variable to use the default" unless $ENV{ENSEMBL_CVS_ROOT_DIR};
+    $schema_patcher = $ENV{ENSEMBL_CVS_ROOT_DIR} . '/ensembl/misc-scripts/schema_patcher.pl';
+}
+die "'$schema_patcher' is not a valid executable" unless -x $schema_patcher;
+
+
+if ($dba->dbc->user eq 'ensro') {
+    warn "Switching to the ensadmin user\n";
+}
```

Comment:
Don't we usually follow the following convention?
```suggestion
warn "Switching to the ensadmin user\n" if ($dba->dbc->user eq 'ensro');
```

Reply:
I wasn't aware this was a convention ...